hs_core/hydroshare/resource.py

```diff
@@ -741,17 +760,54 @@ def publish_resource(pk):
     Exceptions.NotAuthorized - The user is not authorized
     Exceptions.NotFound - The resource identified by pid does not exist
     Exception.ServiceFailure - The service is unable to process the request
+    and other general exceptions
 
-    Note:  This is different than just giving public access to a resource via access control rul
+    Note:  This is different than just giving public access to a resource via access control rule
     """
     resource = utils.get_resource_by_shortkey(pk)
-    resource.raccess.published = True
-    resource.raccess.immutable = True
-    resource.raccess.save()
-    resource.doi = "to be assigned"
-    resource.metadata.create_element('date', type='published', start_date=now().isoformat())
+    # append pending to the doi field to indicate DOI is not activated yet. Upon successful activation, "pending" will be removed from DOI field
+    resource.doi = get_resource_doi(pk, True)
```

Comment:
@pkdash The resource has already been checked for publishing criteria before getting to this view function. Specifically, the publish button does not show up at all on resource landing page before publishing criteria are met, which use the same criteria checking function for making public. So if a user clicks publish button, which invokes this function, at this point it is already ensured publication criteria are met, hence no need to do the check again. Let me know if this clarifies and is sufficient.